File: packages/shared/src/types/studyPlan.ts
export type SeniorityLevel = 'junior' | 'mid' | 'senior' | 'staff' | 'principal';

export interface StudyResource {
  type: 'video' | 'article' | 'book' | 'course';
  title: string;
  url: string;
}

export interface StudyTopic {
  title: string;
  description: string;
  estimated_hours: number;
  resources: StudyResource[];
  practice: string;
  done: boolean;
}

export interface StudyWeek {
  week: number;
  theme: string;
  topics: StudyTopic[];
}

export interface StudyPlan {
  id: string;
  user_id: string;
  role_title: string;
  seniority_level: SeniorityLevel;
  duration_weeks: 1 | 2 | 4 | 8 | 12;
  topics: StudyTopic[];
  progress_pct: number;
  generated_at: string;
  last_updated_at: string;
  weekly_hours_required?: number;
  overview?: string;
  weeks?: StudyWeek[];
  final_project?: string;
  interview_readiness_milestones?: string[];
}

export interface GenerateStudyPlanParams {
  role_title: string;
  seniority_level: SeniorityLevel;
  duration_weeks: 1 | 2 | 4 | 8 | 12;
  current_skills?: string[];
}

File: backend/schemas/study_plan.py
from pydantic import BaseModel


class StudyResource(BaseModel):
    type: str
    title: str
    url: str


class StudyTopic(BaseModel):
    title: str
    description: str
    estimated_hours: int
    resources: list[StudyResource]
    practice: str
    done: bool = False


class StudyWeek(BaseModel):
    week: int
    theme: str
    topics: list[StudyTopic]


class GenerateStudyPlanRequest(BaseModel):
    role_title: str
    seniority_level: str
    duration_weeks: int
    current_skills: list[str] | None = None


class MarkTopicRequest(BaseModel):
    done: bool


class StudyPlanResponse(BaseModel):
    id: str
    user_id: str
    role_title: str
    seniority_level: str
    duration_weeks: int
    topics: list[StudyTopic]
    progress_pct: int
    generated_at: str
    last_updated_at: str
    weekly_hours_required: int | None = None
    overview: str | None = None
    weeks: list[StudyWeek] | None = None
    final_project: str | None = None
    interview_readiness_milestones: list[str] | None = None


class StudyPlanTemplateResponse(BaseModel):
    role: str
    seniority: str
    duration_weeks: int
    overview: str
    weeks: list[StudyWeek]
